ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsTable.tsx

```diff
@@ -0,0 +1,93 @@
+import { Button, ButtonVariant, Tooltip } from '@patternfly/react-core';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+
+import IconText from 'Components/PatternFly/IconText/IconText';
+import TbodyUnified from 'Components/TableStateTemplates/TbodyUnified';
+import { ClusterCheckStatus } from 'services/ComplianceResultsService';
+import { getDistanceStrictAsPhrase } from 'utils/dateUtils';
+import { TableUIState } from 'utils/getTableUIState';
+
+import { coverageClusterDetailsPath } from './compliance.coverage.routes';
+import { getClusterResultsStatusObject } from './compliance.coverage.utils';
+
+export type CheckDetailsTableProps = {
+    currentDatetime: Date;
+    profileName: string;
+    tableState: TableUIState<ClusterCheckStatus>;
+};
+
+function CheckDetailsTable({ currentDatetime, profileName, tableState }: CheckDetailsTableProps) {
+    return (
+        <>
+            <Table>
+                <Thead>
+                    <Tr>
+                        <Th>Cluster</Th>
+                        <Th>Last scanned</Th>
+                        <Th>Compliance status</Th>
+                    </Tr>
+                </Thead>
+                <TbodyUnified
+                    tableState={tableState}
+                    colSpan={3}
+                    errorProps={{
+                        title: 'There was an error loading results for this check',
+                    }}
+                    emptyProps={{
+                        message: 'No results found for this check',
+                    }}
+                    filteredEmptyProps={{
+                        title: 'No results found',
+                        message: 'Clear all filters and try again',
+                    }}
+                    renderer={({ data }) => (
+                        <Tbody>
+                            {data.map((clusterInfo) => {
+                                const {
+                                    cluster: { clusterId, clusterName },
+                                    lastScanTime,
+                                    status,
+                                } = clusterInfo;
+                                const clusterStatusObject = getClusterResultsStatusObject(status);
+                                const firstDiscoveredAsPhrase = getDistanceStrictAsPhrase(
+                                    lastScanTime,
+                                    currentDatetime
+                                );
+
+                                return (
+                                    <Tr key={clusterId}>
+                                        <Td dataLabel="Cluster">
+                                            <Link
+                                                to={generatePath(coverageClusterDetailsPath, {
+                                                    clusterId,
+                                                    profileName,
+                                                })}
```

Comment:
Asking, not necessarily questioning (pardon pun) to confirm my understanding: for a particular check from a profile, this cluster link is to cluster details page that displays all check results from the profile for the cluster?

Thinking ahead to cluster details page, if there is a link for a check from the profile for the cluster, is there any page more specific that this check details page which displays results for all clusters?

Reply:
> Asking, not necessarily questioning (pardon pun) to confirm my understanding: for a particular check from a profile, this cluster link is to cluster details page that displays all check results from the profile for the cluster?

Correct

> Thinking ahead to cluster details page, if there is a link for a check from the profile for the cluster, is there any page more specific that this check details page which displays results for all clusters?

This one made my head hurt but I think I follow. There is not a dedicated page for cluster -> profile -> check, which brings up a good question. Where does that link go? My assumption was it would just take you to the `CheckDetailsPage` for that particular check but I'll confirm with @alanonthegit